tests/Mado/QueryBundle/Objects/QueryBuilderFactoryTest.php

```diff
@@ -109,6 +109,177 @@ public function testFiltersMustContainsAlsoFieldEquality()
             $factory->getAvailableFilters()
         );
     }
+
+    public function testGetFields()
+    {
+        $fields = ['id'];
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setFields($fields);
+        $fieldsReturned = $queryBuilderFactory->getFields();
+
+        $this->assertEquals($fields, $fieldsReturned);
+    }
+
+    /**
+     * @expectedException \RuntimeException
+     */
+    public function testGetFieldsThrowExceptionIfNull()
+    {
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->getFields();
+    }
+
+    public function testSetOrFilters()
+    {
+        $fields = ['id'];
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setOrFilters($fields);
+
+        $this->assertAttributeEquals($fields, 'orFiltering', $queryBuilderFactory);
+    }
+
+    public function testGetOrFilters()
+    {
+        $fields = ['id'];
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setOrFilters($fields);
+
+        $fieldsReturned = $queryBuilderFactory->getOrFilters();
+
+        $this->assertEquals($fields, $fieldsReturned);
+    }
+
+    public function testSetSorting()
+    {
+        $fields = ['id'];
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setSorting($fields);
+
+        $this->assertAttributeEquals($fields, 'sorting', $queryBuilderFactory);
+    }
+
+    public function testGetFilters()
+    {
+        $fields = ['id'];
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->setFilters($fields);
+        $fieldsReturned = $queryBuilderFactory->getFilters();
+
+        $this->assertEquals($fields, $fieldsReturned);
+    }
+
+    /**
+     * @expectedException \RuntimeException
+     */
+    public function testGetQueryBuilderThrowExceptionIfNull()
+    {
+        $queryBuilderFactory = new QueryBuilderFactory($this->manager);
+        $queryBuilderFactory->getQueryBuilder();
+    }
+
+    public function testGetQueryBuilder()
```

Comment:
this test isnt necessary: to assert that returned object is of a specific type we just need php's type hinting.